roles/kubernetes-apps/policy_controller/calico/tasks/main.yml

```diff
@@ -2,19 +2,21 @@
 - name: Set cert dir
   set_fact:
     calico_cert_dir: "{{ canal_cert_dir }}"
-  when: kube_network_plugin == 'canal'
+  when:
+    - kube_network_plugin == 'canal'
   tags:
     - facts
     - canal
 
-- name: Delete the old calico-policy-controller if it exist
+- name: Delete the old calico-kube-controllers deployment if it exist
   kube:
-    name: calico-policy-controller
+    name: calico-kube-controllers
     kubectl: "{{bin_dir}}/kubectl"
-    resource: rs
+    resource: deployment
```

Comment:
change it back. the deployment did not change names, so this should not purge the deployment calico-kube-controllers